### PR TITLE
C#/Go/Java/JS/Python/Ruby: Update the description and qhelp of the Zipslip query

### PR DESCRIPTION
**Pull request by atorralba**
All filesystem operations, not just writes, with paths built from untrusted archive entry names are dangerous.

### CODE REVIEW

**Review by RasmusWL — APPROVED** [review]
Python :+1:

**Review by michaelnebel — DISMISSED** [review]
C# 👍

**Review by owen-mc — APPROVED** [review]
Go LGTM

**Comment by mchammer01** [comment]
I'll review this on behalf of docs.

File: go/ql/src/change-notes/2023-06-16-zipslip-rename.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+---
+category: fix
+---
+* The query "Arbitrary file write during zip extraction ("zip slip")" (`go/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")".
```
Comment (mchammer01) [inline go/ql/src/change-notes/2023-06-16-zipslip-rename.md:4]: As per the punctuation rules from the Microsoft and the GitHub style guides
```suggestion
* The query "Arbitrary file write during zip extraction ("zip slip")" (`go/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")."
```

File: java/ql/src/change-notes/2023-06-16-zipslip-rename.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+category: fix
+---
+* The query "Arbitrary file write during archive extraction ("Zip Slip")" (`java/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")".
```
Comment (mchammer01) [inline java/ql/src/change-notes/2023-06-16-zipslip-rename.md:4]: As per the punctuation rules from the Microsoft and the GitHub style guides
```suggestion
* The query "Arbitrary file write during archive extraction ("Zip Slip")" (`java/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")."
```

File: python/ql/src/change-notes/2023-06-16-zipslip-rename.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+category: fix
+---
+* The query "Arbitrary file write during archive extraction ("Zip Slip")" (`py/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")".
```
Comment (mchammer01) [inline python/ql/src/change-notes/2023-06-16-zipslip-rename.md:4]: As per the punctuation rules from the Microsoft and the GitHub style guides
```suggestion
* The query "Arbitrary file write during archive extraction ("Zip Slip")" (`py/zipslip`) has been renamed to "Arbitrary file access during archive extraction ("Zip Slip")."
```

**Review by mchammer01 — APPROVED** [review]
@atorralba - this LGTM ✨ 
A few minor editorial comments for your consideration. Feel free to ignore the comments you don't agree with.

File: csharp/ql/src/Security Features/CWE-022/ZipSlip.qhelp [hunk 4]
```diff
@@ -3,16 +3,16 @@
   "qhelp.dtd">
 <qhelp>
 <overview>
-<p>Extracting files from a malicious zip archive without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+<p>Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply csharp/ql/src/Security Features/CWE-022/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: go/ql/src/Security/CWE-022/ZipSlip.qhelp [hunk 5]
```diff
@@ -5,18 +5,18 @@
 
 <overview>
 <p>
-Extracting files from a malicious zip archive without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply go/ql/src/Security/CWE-022/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: java/ql/src/Security/CWE/CWE-022/ZipSlip.qhelp [hunk 6]
```diff
@@ -3,17 +3,16 @@
   "qhelp.dtd">
 <qhelp>
 <overview>
-<p>Extracting files from a malicious zip archive (or another archive format)
-without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+<p>Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply java/ql/src/Security/CWE/CWE-022/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: javascript/ql/src/Security/CWE-022/ZipSlip.qhelp [hunk 7]
```diff
@@ -4,16 +4,16 @@
 <qhelp>
 
 <overview>
-<p>Extracting files from a malicious zip archive without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+<p>Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply javascript/ql/src/Security/CWE-022/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: python/ql/src/experimental/Security/CWE-022/ZipSlip.qhelp [hunk 8]
```diff
@@ -4,16 +4,16 @@
 <qhelp>
 
 <overview>
-<p>Extracting files from a malicious zip archive without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+<p>Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply python/ql/src/experimental/Security/CWE-022/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: ruby/ql/src/experimental/cwe-022-zipslip/ZipSlip.qhelp [hunk 9]
```diff
@@ -4,16 +4,16 @@
 <qhelp>
 
 <overview>
-<p>Extracting files from a malicious tar archive without validating that the destination file path
-is within the destination directory can cause files outside the destination directory to be
-overwritten, due to the possible presence of directory traversal elements (<code>..</code>) in
+<p>Accessing filesystem paths built from the name of an archive entry without validating that the
```
[the remark this answers is not in the document]
Comment (atorralba) [reply ruby/ql/src/experimental/cwe-022-zipslip/ZipSlip.qhelp]: Applied @asgerf's suggestion instead, as you recommended.

File: csharp/ql/src/Security Features/CWE-022/ZipSlip.ql [hunk 10]
```diff
@@ -1,8 +1,8 @@
 /**
- * @name Arbitrary file write during zip extraction ("Zip Slip")
- * @description Extracting files from a malicious zip archive without validating that the
- *              destination file path is within the destination directory can cause files outside
- *              the destination directory to be overwritten.
+ * @name Arbitrary file access during archive extraction ("Zip Slip")
```
[the remark this answers is not in the document]
Comment (atorralba) [reply csharp/ql/src/Security Features/CWE-022/ZipSlip.ql:2]: Adjusted using a combination of your and @asgerf's recommendations in 8f6d2ed2f9045420e8331852f1b7e1942d9775fc.

File: java/ql/src/Security/CWE/CWE-022/ZipSlip.ql [hunk 11]
```diff
@@ -1,8 +1,8 @@
 /**
- * @name Arbitrary file write during archive extraction ("Zip Slip")
- * @description Extracting files from a malicious archive without validating that the
- *              destination file path is within the destination directory can cause files outside
- *              the destination directory to be overwritten.
+ * @name Arbitrary file access during archive extraction ("Zip Slip")
+ * @description Accessing filesystem paths built from the name of an archive entry without
```
[the remark this answers is not in the document]
Comment (atorralba) [reply java/ql/src/Security/CWE/CWE-022/ZipSlip.ql]: Adjusted using a combination of your and @asgerf's recommendations in https://github.com/github/codeql/commit/8f6d2ed2f9045420e8331852f1b7e1942d9775fc.

File: python/ql/src/experimental/Security/CWE-022/ZipSlip.ql [hunk 12]
```diff
@@ -1,8 +1,8 @@
 /**
- * @name Arbitrary file write during archive extraction ("Zip Slip")
- * @description Extracting files from a malicious archive without validating that the
- *              destination file path is within the destination directory can cause files outside
- *              the destination directory to be overwritten.
+ * @name Arbitrary file access during archive extraction ("Zip Slip")
+ * @description Accessing filesystem paths built from the name of an archive entry without
```
[the remark this answers is not in the document]
Comment (atorralba) [reply python/ql/src/experimental/Security/CWE-022/ZipSlip.ql]: Adjusted using a combination of your and @asgerf's recommendations in https://github.com/github/codeql/commit/8f6d2ed2f9045420e8331852f1b7e1942d9775fc.

**Comment by atorralba** [comment]
Thanks @asgerf and @mchammer01 for the suggestions! Let me know if anything else can still be improved.

**Review by mchammer01 — APPROVED** [review]
@atorralba - thanks for addressing my comments, LGTM 🚀

**Review by asgerf — APPROVED** [review]
👍

**Pull request merged**